ExecutionException is thrown while making lightweight tracks secondary

After I make lightweight tracks primary, when I try to make other lightweight tracks as secondary I get 
org.eclipse.core.commands.ExecutionException thrown on the log. But the track is assigned as secondary track.
### Steps to reproduce:
1. Load Lightweight_Test_Tracks.rep
2. Expand red_tracks layer
3. Right click the first track (Fisherone_002) > Make Primary
4. Right click the second track (Fishertwo_003) > Make Secondary
This time it logs the below exception 
Note:
This issue is reproducible in the previous releases of Debrief Version: 3.0.494 Build id: 20191025

#### StackTrace
org.eclipse.core.commands.ExecutionException: While executing the operation, an exception occurred
	at org.eclipse.core.commands.operations.DefaultOperationHistory.execute(DefaultOperationHistory.java:503)
	at org.mwc.cmap.core.CorePlugin.run(CorePlugin.java:825)
	at org.mwc.debrief.core.editors.PlotOutlinePage$21.run(PlotOutlinePage.java:1583)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:580)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:412)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4131)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1056)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3944)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3547)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:635)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:660)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:597)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1468)
Caused by: java.lang.ClassCastException: Debrief.Wrappers.Track.LightweightTrackWrapper cannot be cast to Debrief.Wrappers.ISecondaryTrack
	at org.mwc.debrief.track_shift.views.BaseStackedDotsView.getTargetZones(BaseStackedDotsView.java:2479)
	at org.mwc.debrief.track_shift.views.BaseStackedDotsView.updateTargetZones(BaseStackedDotsView.java:3554)
	at org.mwc.debrief.track_shift.views.BaseStackedDotsView.access$8(BaseStackedDotsView.java:3543)
	at org.mwc.debrief.track_shift.views.BaseStackedDotsView$3.tracksUpdated(BaseStackedDotsView.java:664)
	at org.mwc.cmap.core.DataTypes.TrackData.TrackManager.setPrimaryImpl(TrackManager.java:567)
	at org.mwc.cmap.core.DataTypes.TrackData.TrackManager.setPrimary(TrackManager.java:546)
	at org.mwc.debrief.core.editors.PlotOutlinePage$21$1.doItTo(PlotOutlinePage.java:1546)
	at org.mwc.debrief.core.editors.PlotOutlinePage.applyOperation(PlotOutlinePage.java:379)
	at org.mwc.debrief.core.editors.PlotOutlinePage$SelectionOperation.execute(PlotOutlinePage.java:287)
	at org.eclipse.core.commands.operations.DefaultOperationHistory.execute(DefaultOperationHistory.java:495)

--- a/org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/views/BaseStackedDotsView.java
+++ b/org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/views/BaseStackedDotsView.java
@@ -2476,54 +2476,59 @@ abstract public class BaseStackedDotsView extends ViewPart implements
           .getSecondaryTracks();
       if (secTracks != null && secTracks.length == 1)
       {
-        final ISecondaryTrack sw = (ISecondaryTrack) secTracks[0];
-        Enumeration<Editable> iter = sw.segments();
-
-        // have a peak, to check if we've actually got a segment list
-        if (iter.hasMoreElements())
+        // check it's a secondary track
+        WatchableList nextT = secTracks[0];
+        if (nextT instanceof ISecondaryTrack)
         {
-          final Editable first = iter.nextElement();
-          if (first instanceof SegmentList)
-          {
-            final SegmentList segs = (SegmentList) first;
-            iter = segs.elements();
-          }
-        }
+          final ISecondaryTrack sw = (ISecondaryTrack) nextT;
+          Enumeration<Editable> iter = sw.segments();
 
-        while (iter.hasMoreElements())
-        {
-          final TrackSegment thisSeg = (TrackSegment) iter.nextElement();
-          if (thisSeg instanceof RelativeTMASegment)
+          // have a peak, to check if we've actually got a segment list
+          if (iter.hasMoreElements())
           {
-            // do we have a first color?
-            final Editable firstElement = thisSeg.elements().nextElement();
-            final Color color;
-            if (firstElement != null)
+            final Editable first = iter.nextElement();
+            if (first instanceof SegmentList)
             {
-              final FixWrapper fix = (FixWrapper) firstElement;
-              color = fix.getColor();
+              final SegmentList segs = (SegmentList) first;
+              iter = segs.elements();
             }
-            else
-            {
-              color = Color.RED;
-            }
-
-            final RelativeTMASegment rel = (RelativeTMASegment) thisSeg;
-            final Zone newZ = new Zone(rel.getDTG_Start().getDate().getTime(),
-                rel.getDTG_End().getDate().getTime(), color);
-            zones.add(newZ);
           }
-          else if (thisSeg instanceof TrackSegment)
+
+          while (iter.hasMoreElements())
           {
-            final TrackSegment seg = thisSeg;
-            if (!thisSeg.isEmpty())
+            final TrackSegment thisSeg = (TrackSegment) iter.nextElement();
+            if (thisSeg instanceof RelativeTMASegment)
             {
-              final FixWrapper firstE = (FixWrapper) thisSeg.elements()
-                  .nextElement();
-              final Color color = firstE.getColor();
-              final Zone newZ = new Zone(seg.startDTG().getDate().getTime(), seg
-                  .endDTG().getDate().getTime(), color);
+              // do we have a first color?
+              final Editable firstElement = thisSeg.elements().nextElement();
+              final Color color;
+              if (firstElement != null)
+              {
+                final FixWrapper fix = (FixWrapper) firstElement;
+                color = fix.getColor();
+              }
+              else
+              {
+                color = Color.RED;
+              }
+
+              final RelativeTMASegment rel = (RelativeTMASegment) thisSeg;
+              final Zone newZ = new Zone(rel.getDTG_Start().getDate().getTime(),
+                  rel.getDTG_End().getDate().getTime(), color);
               zones.add(newZ);
+            }
+            else if (thisSeg instanceof TrackSegment)
+            {
+              final TrackSegment seg = thisSeg;
+              if (!thisSeg.isEmpty())
+              {
+                final FixWrapper firstE = (FixWrapper) thisSeg.elements()
+                    .nextElement();
+                final Color color = firstE.getColor();
+                final Zone newZ = new Zone(seg.startDTG().getDate().getTime(),
+                    seg.endDTG().getDate().getTime(), color);
+                zones.add(newZ);
+              }
             }
           }
         }

--- a/org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/views/BaseStackedDotsView.java
+++ b/org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/views/BaseStackedDotsView.java
@@ -1026,11 +1026,6 @@ abstract public class BaseStackedDotsView extends ViewPart implements
    * {@link ValueAxis#setStandardTickUnits(TickUnitSource)} method inherited from the
    * {@link ValueAxis} class).
    *
-   * @param zone
-   *          the time zone (<code>null</code> not permitted).
-   * @param locale
-   *          the locale (<code>null</code> not permitted).
-   *
    * @return A collection of standard date tick units.
    *
    * @since 1.0.11
@@ -1308,7 +1303,6 @@ abstract public class BaseStackedDotsView extends ViewPart implements
     setZoneChartsVisible(_showZones.isChecked());
   }
 
-  @SuppressWarnings("static-method")
   protected Runnable getDeleteAmbiguousCutsOperation()
   {
     // ditch, let the child class(es) override it
@@ -2280,7 +2274,7 @@ abstract public class BaseStackedDotsView extends ViewPart implements
   /**
    * format the value in a suitable way for marking the current value of the cursor
    *
-   * @param current
+   * @param value
    *          data value at cursor
    * @return suitably formatted version
    */
@@ -2317,7 +2311,6 @@ abstract public class BaseStackedDotsView extends ViewPart implements
    *
    * @return
    */
-  @SuppressWarnings("static-method")
   protected Runnable getDeleteCutsOperation()
   {
     return null;
@@ -2349,7 +2342,6 @@ abstract public class BaseStackedDotsView extends ViewPart implements
    *
    * @return
    */
-  @SuppressWarnings("static-method")
   protected Runnable getResolveAmbiguityOperation()
   {
     return null;
@@ -2612,9 +2604,15 @@ abstract public class BaseStackedDotsView extends ViewPart implements
       }
       else
       {
-        // create a leg from the previous end to this start
-        legs.add(new Zone(lastZig.getEnd(), zig.getStart(), randomProv
-            .getZoneColor()));
+        // technically we don't need this null check, the logic
+        // _should_ ensure it's non-null. But, this will stop the 
+        // complainer moaning
+        if(lastZig != null)
+        {
+          // create a leg from the previous end to this start
+          legs.add(new Zone(lastZig.getEnd(), zig.getStart(), randomProv
+              .getZoneColor()));
+        }
       }
 
       // remember the zig
@@ -3258,11 +3256,11 @@ abstract public class BaseStackedDotsView extends ViewPart implements
    * slice the target bearings according to these zones
    *
    * @param ownshipZones
+   * @param cuts
    * @param randomProv
+   * @param tgtTrack
    * @param slicePrecision
-   * @param secondaryTrack
-   * @param targetBearingSeries2
-   * @return
+   * @return list of zones
    */
   protected List<Zone> sliceTarget(final List<Zone> ownshipZones,
       final List<SensorContactWrapper> cuts, final ColorProvider randomProv,
@@ -3552,13 +3550,9 @@ abstract public class BaseStackedDotsView extends ViewPart implements
       // clear the zone charts, but maybe not the primary
       clearZoneCharts(false, true, false);
 
-      // do we have a target zone chart?
-      if (targetZoneChart != null)
-      {
-        // initialise the zones
-        final List<Zone> zones = getTargetZones();
-        targetZoneChart.setZones(zones);
-      }
+      // initialise the zones
+      final List<Zone> zones = getTargetZones();
+      targetZoneChart.setZones(zones);
     }
   }
 

